Add reference to tutorial page in cli (#3267)
optuna/optuna: docs/source/reference/cli.rst

The :mod:`~optuna.cli` module implements Optuna's command-line functionality using the `cliff framework <https://docs.openstack.org/cliff/latest/index.html>`_.

.. seealso::

>    Examples are provided in the :ref:`cli` tutorial.

Reviewer: I guess the tutorial page does not show an example but also shows use cases. How about rephrasing it, for example, as follows?
```suggestion
    The :ref:`cli` tutorial provides use-cases with examples.
```
